weed/util/fullpath.go

```diff
@@ -64,19 +44,18 @@ func (fp FullPath) AsInode(unixTime int64) uint64 {
 }
 
 func Join(names ...string) string {
-	newName := clearName(filepath.ToSlash(filepath.Join(names...)))
-	if newName != "/" && strings.HasSuffix(filepath.ToSlash(strings.Join(names, "")), "/") {
-		newName += "/"
-	}
-	return newName
+	return clearName(path.Join(names...))
 }
 
 func clearName(name string) string {
 	name = strings.ToValidUTF8(name, "?")
-	slashed := strings.HasSuffix(name, "/")
-	name = filepath.Clean(name)
-	if name != "/" && slashed {
-		name += "/"
+	name = strings.ReplaceAll(name, "\\", "/")
+	name = path.Clean(name)
+	if name == "." {
+		name = "/"
```

Comment:
this seems wrong.

Reply:
Yes, it should add `/` prefix before clean the path.